Intersect (Core)/Threading/Threaded`1.cs

```diff
@@ -1,33 +1,32 @@
-﻿using System;
-
-namespace Intersect.Threading
+﻿namespace Intersect.Threading
 {
-
-    /// <inheritdoc />
+    /// <summary>
+    /// Represents a generic base class for creating and managing new threads.
+    /// </summary>
+    /// <typeparam name="TArgument">Type of argument that will be passed to a new thread.</typeparam>
     public abstract partial class Threaded<TArgument> : Threaded
     {
-
-        /// <inheritdoc />
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Threaded{TArgument}"/> class.
+        /// </summary>
+        /// <param name="name">Name of new thread.</param>
         protected Threaded(string name = null) : base(name)
         {
         }
 
-        /// <inheritdoc />
+        /// <summary>
+        /// Overrides the <see cref="Threaded.ThreadStart"/> method.
+        /// </summary>
         protected override void ThreadStart(params object[] args)
         {
-            // TODO: Generic utility that checks arg types against expected types
-            if (args == null || args.Length < 1)
-            {
-                throw new ArgumentOutOfRangeException(
-                    nameof(args), $@"Expected one argument of type {typeof(TArgument).FullName}."
-                );
-            }
-
-            ThreadStart((TArgument) args[0]);
+            ArgumentTypeChecker<TArgument>.CheckArgumentTypes(args);
+            ThreadStart((TArgument)args[0]);
         }
 
+        /// <summary>
+        /// Main entry point for a new thread, where the argument of type <typeparamref name="TArgument"/> is passed.
+        /// </summary>
+        /// <param name="argument">Argument of type <typeparamref name="TArgument"/> that will be passed to a new thread.</param>
```

Comment:
This does not provide any meaningful information to the reader, it is just restating what the code clearly does.